specs/trip_dispatch_spec.rb

```diff
@@ -111,4 +111,55 @@
       expect(passenger.trips).must_include trip
     end
   end
+
+  describe "#request_trip" do
+    before do
+            @dispatcher = RideShare::TripDispatcher.new(USER_TEST_FILE, TRIP_TEST_FILE, DRIVER_TEST_FILE)
+          end
+    it "returns instance of Trip" do
+      expect(@dispatcher.request_trip(1)).must_be_kind_of RideShare::Trip
+
+    end
+
+    it "increases TripDispatcher-@trips" do
+      before = @dispatcher.trips.length
+      trip = @dispatcher.request_trip(1)
+      after = before + 1
+      expect(@dispatcher.trips.length).must_equal after
+    end
+
+    # Skipped because couldn't apply above test logic to this test in time to submit
+    xit "increases Driver-@driven_trips" do
+      @driver = RideShare::Driver.new(id: 5, name: "Driver5",
+        vin: "1B6CF40K1J3Y74UY2",
+        phone: '111-111-1115',
+        status: :AVAILABLE)
+      before = @driven_trips.length
+      @dispatcher.request_trip(1)
+      after = before + 1
+      expect(@driven_trips.length).must_equal after
+
+    end
+  # Skipped because couldn't apply TripDipatcher-@trips test logic to this test in time to submit
+    xit "increases User-@trips" do
+      @user = RideShare::User.new(id: 9, name: "Merl Glover III",
+        phone: "1-602-620-2330 x3723", trips: [])
+      before = @user.trips.length
+      trip = @dispatcher.request_trip(9)
+      after = before + 1
+      expect(@user.trips.length).must_equal after
+    end
+
+    it "selects first :AVAILABLE driver" do
+      trip = @dispatcher.request_trip(1)
+      expect(trip.driver.id).must_equal 5
```

Comment:
This is a solid test! We could expand on it by requesting a second trip after the first one, and then verifying that we got the next available driver (and not driver #5 a second time).